src/log.rs

```diff
@@ -0,0 +1,132 @@
+use std::error::Error;
+use std::fs::{File, OpenOptions};
+use std::io::{self, BufRead, BufReader, BufWriter, Write};
+use std::path::Path;
+
+/// Represents a single log entry with an index, start time, message, and elapsed time.
+#[derive(Debug)]
+pub struct LogEntry {
+    pub index: usize,         // Index of the log entry
+    pub start_time: String,   // Start time of the log entry
+    pub message: String,      // Message associated with the log entry
+    pub elapsed_time: String, // Elapsed time recorded in the log entry
+}
+
+/// Reads logs from a specified file and returns a vector of `LogEntry`.
+///
+/// # Arguments
+/// - `file_path`: The path to the log file.
+///
+/// # Returns
+/// - `Ok(Vec<LogEntry>)`: A vector of log entries if successful.
+/// - `Err(std::io::Error)`: An error if file operations fail.
+pub fn read_logs_from_file(file_path: &str) -> Result<Vec<LogEntry>, std::io::Error> {
```

Comment:
Maybe better use some other format for log activity, like CSV, jsonx etc.? It is simplify in general logic for read/write/delete in this file, I think